proto/src/constraints.proto

```diff
@@ -59,6 +61,7 @@ message ValueConstraintMsg {
   oneof second_field {
     double value = 2;
     string regex_pattern = 5;
+    google.protobuf.ListValue value_set = 6;
```

Comment:
Can you explain a little why you chose ListValue as the type here? What were the other options considered? I see that you can support mixed sets of strings and numeric types in your tests. What does the protobuf serialization look like?

Reply:
I chose ListValue as the type to be able to support mixed value sets, because it's a wrapper over a repeated field of arbitraty values. Since this value is stored as a set in the code, there won't be any duplicate values in the protobuf serialization. Other options were to use repeated string / int / float, etc., but these would only support sets of values with only one type. I didn't find an implementation for serializing a set instead of a list.

The protobuf serialization looks like this:

name: "value IN {1, \'abc\', 2}"
op: IN
value_set {
  values {
    list_value {
      values {
        number_value: 1.0
      }
      values {
        string_value: "abc"
      }
      values {
        number_value: 2.0
      }
    }
  }
}